lua/neo-tree/sources/filesystem/commands.lua

```diff
@@ -222,14 +222,21 @@ M.rename = function(state)
 end
 
 M.set_root = function(state)
-  local tree = state.tree
-  local node = tree:get_node()
-  if node.type == "directory" then
-    if state.search_pattern then
-      fs.reset_search(state, false)
-    end
-    fs._navigate_internal(state, node.id, nil, nil, false)
+  if state.search_pattern then
+    fs.reset_search(state, false)
   end
+
+  local node = state.tree:get_node()
+  if not node then
+    return
+  end
+
+  local id = node.type == "directory" and node:get_id() or node:get_parent_id()
```

Comment:
Sorry to be late to the party, but this is not quite right because it assumes there are only files and folders, but there may also be nested files. To get a parent folder via navigating up nodes, it needs to be a while loop which tests for `node.type == "directory"` in each loop and has some safety bail out to prevent an infinite loop.

The alternative is to get the parent by splitting the filename/path with `utils.split_path`.

Reply:
Hi, Thank you for your comment, but i actually have no idea what nested files are like. So could you please provide some specific example of `nested files`?

Reply:
Yes, that's perfect.

BTW - this is the nested file feature: https://github.com/nvim-neo-tree/neo-tree.nvim/issues/118